test/rules/src/https_everywhere_checker/http_client.py

```diff
@@ -379,13 +377,11 @@ def fetchHtml(self, url):
 			if httpCode == 0:
 				raise HTTPFetcherError("Pycurl fetch failed for '%s'" % newUrl)
 			elif httpCode in (301, 302, 303, 307):
-				# Parse out the headers and extract location, case-insensitively.
-				# If there are multiple location headers, pick the last one.
-				headers = dict()
-				for k, v in self._headerRe.findall(headerStr):
-					headers[k.lower()] = v
-				location = headers.get('location')
-				if not location:
+				location = None
+				for piece in headerStr.split('\n'):
+					if piece.startswith('Location:'):
```

Comment:
We should be careful here.  Although the `Location` header is usually capitalized in this way, it is not always.  See https://www.shodan.io/search?query=location%3A